crypto/x509/punycode.c

```diff
@@ -0,0 +1,498 @@
+/*
+punycode.c from RFC 3492
```

Comment:
This file needs a bit of work to bring it into line with the OpenSSL style, e.g. comment style
I think we will need an OMC discussion with regards to the licence associated with this code. AFAICT it should be ok, but it's not 100% clear.

Reply:
It's the most important point, I think. Linking to some IDN-processing lib seems a bad idea, and I'm not sure that I can write this code better than it is written now.

Reply:
I initiated a discussion on openssl-project.

Reply:
From a style viewpoint, I'd personally like to see the file content remain essentially as-is, so it can be compared to match the code in the RFC.  We should not be editing this file, other than some license-related text at the top.

Reply:
> From a style viewpoint, I'd personally like to see the file content remain essentially as-is, so it can be compared to match the code in the RFC.

Strongly disagree. We've worked hard to get a consistent style across the OpenSSL codebase and I for one do not want to go back to the days where we had a different style in every file. The changes required to bring this into line are not that great.

Reply:
I guess my perspective is that this is a curated external artefact, in lieu of an external library dependency. That is, this is code we're importing, but not maintaining independently of the upstream. We're also not providing a Punycode interface to applications, but just using it internally.

Mind you this not a rigid view, just my immediate take.

Reply:
As far as I can see, some Travis failures are related to this code, so we can't leave it unchanged.

Reply:
> I guess my perspective is that this is a curated external artefact, in lieu of an external library dependency.

I don't see it that way at all. Once incorporated we will very likely want to make changes and tweaks over time. This will become OpenSSL code and we should treat it as such.